File: app/src/test/java/com/cdp/remote/presentation/screen/scheduler/SchedulerModelOptionsTest.kt
package com.cdp.remote.presentation.screen.scheduler

import org.junit.Assert.assertEquals
import org.junit.Assert.assertTrue
import org.junit.Test

class SchedulerModelOptionsTest {
    @Test
    fun `codex exposes codex model options`() {
        val options = schedulerModelOptionsForIde("Codex")

        assertEquals("默认", options.first().label)
        assertTrue(options.any { it.value == "GPT-5.5" })
        assertTrue(options.any { it.value == "Extra High" })
    }

    @Test
    fun `cursor exposes cursor model options`() {
        val options = schedulerModelOptionsForIde("Cursor")

        assertTrue(options.any { it.value == "Composer 2" })
        assertTrue(options.any { it.value == "Sonnet" })
    }

    @Test
    fun `uitty only exposes default because it has no global model switcher`() {
        val options = schedulerModelOptionsForIde("uitty")

        assertEquals(listOf(SchedulerModelOption("默认", "")), options)
    }

    @Test
    fun `dynamic ide models replace local presets`() {
        val options = schedulerModelOptionsForIde(
            ideName = "DSME",
            liveModels = listOf("Gemini 3.1 Pro (High)", "Claude Sonnet 4.6 (Thinking)")
        )

        assertEquals("默认", options.first().label)
        assertTrue(options.any { it.value == "Gemini 3.1 Pro (High)" })
        assertTrue(options.any { it.value == "Claude Sonnet 4.6 (Thinking)" })
        assertTrue(options.none { it.value == "MiniMax" })
    }
}

File: relay-dev/scheduler_model_options.test.js
const assert = require('assert');
const {
    defaultSchedulerModelOptions,
    sanitizeModelOptions
} = require('./scheduler_model_options');

assert.deepStrictEqual(
    sanitizeModelOptions([' Gemini 3.1 Pro ', '', 'Gemini 3.1 Pro', null, 'Claude Sonnet 4.6', '✓ deepseek-v4-flash']),
    ['Gemini 3.1 Pro', 'Claude Sonnet 4.6', 'deepseek-v4-flash']
);

assert.ok(defaultSchedulerModelOptions('DSME').includes('Gemini 3.1 Pro'));
assert.ok(defaultSchedulerModelOptions('Codex').includes('GPT-5.5'));
assert.deepStrictEqual(defaultSchedulerModelOptions('uitty:26954'), []);

console.log('scheduler_model_options tests passed');
